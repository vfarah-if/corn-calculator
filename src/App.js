import React from "react";

import TripCalculator from "./components/TripCalculator";
import Header from "./components/Header";

import "./App.css";

function App() {
  return (
    <div className="App">
      <Header title="River trip adviser"/>
      <div className="App-content">
        <TripCalculator></TripCalculator>
      </div>
    </div>
  );
}

export default App;
