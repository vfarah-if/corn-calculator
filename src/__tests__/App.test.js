import { render, screen } from '@testing-library/react';

import App from '../App';

describe('App', () => {
  test('should contain a header with river trip adviser', () => {
    render(<App />);
    const linkElement = screen.getByText(/River trip adviser/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('should contain a corn quantity label', () => {
    render(<App />);
    const linkElement = screen.getByText(/Corn Quantity:/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('should contain two quantity inputs', () => {
    const component = render(<App />);
    
    const inputElements = component.container.querySelectorAll('input[name="quantity"]');

    expect(inputElements.length).toBe(2);
  });

  test('should contain corn quantity with initial default value of 0', () => {
    const component = render(<App />);
    
    const inputElements = component.container.querySelectorAll('input[name="quantity"]');
    const inputElement = inputElements[0];

    expect(inputElement).toBeTruthy();
    expect(inputElement.value).toBe("0");
  });

  test('should contain geese quantity with initial default value of 0', () => {
    const component = render(<App />);
    
    const inputElements = component.container.querySelectorAll('input[name="quantity"]');
    const inputElement = inputElements[1];

    expect(inputElement).toBeTruthy();
    expect(inputElement.value).toBe("0");
  });

  test('should calculate the cost of 0 bag of corn and 0 geese @ £ 0.25', () => {
    render(<App />);
    
    const linkElement = screen.getByText(/£ 0.25/i);
    
    expect(linkElement).toBeInTheDocument();
  });

  test(`should default user message to "no cargo to take"`, () => {
    render(<App />);
    
    const linkElement = screen.getByText(/no cargo to take/i);
    
    expect(linkElement).toBeInTheDocument();
  });

  test('should get default app in the expected html format', () => {
    const component = render(<App />);
          
    expect(component.container).toMatchSnapshot();
  });

});
