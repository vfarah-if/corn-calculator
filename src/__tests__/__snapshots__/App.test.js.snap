// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should get default app in the expected html format 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
      style="background-image: url(/farm.jpg); background-position: center; background-size: cover; background-repeat: no-repeat;"
    >
      <h1>
        River trip adviser
      </h1>
    </header>
    <div
      class="App-content"
    >
      <form>
        <legend>
          Plan your trip
        </legend>
        <fieldset>
          <div
            class="input-group"
          >
            <label
              forhtml="quantity"
            >
              Corn Quantity
              :
            </label>
            <input
              min="0"
              name="quantity"
              type="number"
              value="0"
            />
          </div>
          <div
            class="input-group"
          >
            <label
              forhtml="quantity"
            >
              Geese Quantity
              :
            </label>
            <input
              min="0"
              name="quantity"
              type="number"
              value="0"
            />
          </div>
        </fieldset>
        <label>
          Â£ 
          0.25
        </label>
        <p
          class="user-info"
        >
          no cargo to take
        </p>
        <p
          class="user-info"
        >
          Trip Count: 
          1
        </p>
      </form>
    </div>
  </div>
</div>
`;
